### ISSUE TITLE
Managed Identity Wallet Release 23.12 Documentation Acceptance Criteria

### ISSUE THREAD

**Issue by kelaja** [eclipse-tractusx/sig-release#102]
# Release Documentation 23.12

Source in Catena-X Confluence and Expert Contacts [here](https://confluence.catena-x.net/x/DOZkBQ) (Source only accessible for Catena-X Consortia members in current transition phase).

- [x] **Source Code** https://github.com/eclipse-tractusx/managed-identity-wallet/issues/148

  - Code is centrally managed in Eclipse Tractus-X repository.
  - All active and relevant [Release Guidelines](https://eclipse-tractusx.github.io/docs/release) are fulfilled (with confirmation by DevSecOps SystemTeam).

  _Artefact Repo:_
    - [Eclipse Tractus-X](https://github.com/eclipse-tractusx)
    - FOSS [Infopage](https://confluence.catena-x.net/x/_AZHAw)(Source only accessible for Catena-X Consortia members in current transition phase)
  
  _Best Practice_
    - Consult the [regular office hours](https://catenax-ng.github.io/) as early as possible; expect optimization loops based on feedback.

  _Contact:_
    - DevSecOps System Team
  
- [x] **Architecture Documents**

  - Arc42 documentation completed for relevant product version.
  - In MarkDown format. Link to document available. Must point to leading repository within Tractus-X. Process the JIRA ticket and schedule an appointment with Expert to obtain approval prior to Gate review.
  - Provide link to documentation as early as possible and _mark changes_ to previous version (if applicable); expect optimization loops based on feedback.
  
  _Artefact Repo:_
    - [GitHub Readme how-to](https://confluence.catena-x.net/x/iVIAAQ) (Source only accessible for Catena-X Consortia members in current transition phase)

  _Contact:_
    - SYSTEM ARCHITECT

- [x] **Administrator`s Guide** (User assistance)

  - Admin Guide is available with the software at the same time
  - Admin Guide is correct and up to date; english is a must
  - The documentation is of appropriate maturity for any Operating Company with global business practice intentions.
  - In MarkDown format. Link to document available. Must point to leading repository within Tractus-X.
  
  _Best Practice_:

    - Process the issue and schedule an appointment with SYSTEM ARCHITECT to obtain approval prior to Gate review.
    - Provide link to documentation as early as possible and mark changes to previous version (if applicable); expect optimization loops based on feedback.
    - User assistance ensures that for example administrators get all the information they need to accomplish their tasks with the software. Refer to a administration guide, which covers "install/deploy, configure the software" – as appropriate for the type of software and the information needs of the target group(s)
    - UI text and embedded help complete the User assistance.

  _Contact:_
    - SYSTEM ARCHITECT

- [x] **End-User Manual** (User assistance)
  - End-User Manual is available with the software at the same time
  - End-User Manual is correct and up to date; english is a must
  - The documentation is of appropriate maturity to be handed over from the CX Consortia to any Operations Company with global business practice intentions.

  _Best Practice:_

    - Process the issue and schedule an appointment with SYSTEM ARCHITECT to obtain approval prior to Gate review.
    - Provide link to documentation as early as possible and mark changes to previous version (if applicable); expect optimization loops based on feedback.
    - User assistance ensures that end users and others get all the information they need to accomplish their tasks with the software. Refer to a user guide, which covers "install/deploy, configure, and use the software" – as appropriate for the type of software and the information needs of the target group(s)
    - UI text and embedded help complete the User assistance.

  _Contact:_
    - SYSTEM ARCHITECT

- [x] **Interfaces Documentation**
  - API documentation contains all relevant interfaces for integration testing and is completed for relevant product version.
  - Link to document available.
  - Interface contract signed by all involved parties.

  _Best Practice:_
  
    - Process the issue and schedule an appointment with SYSTEM ARCHITECT to obtain approval prior to Gate review.
    - Provide link to documentation as early as possible and mark changes to previous version (if applicable); expect optimization loops based on feedback.
    - see [Open APIs](https://www.openapis.org/)

  _Contact:_ 
    - SYSTEM ARCHITECT

- [x] **UX consistency** Style Guideline for User Interfaces
  
    - Mandatory for FrontEnd modules where the IP is Open Source or owned by Catena-X.
    - User Interfaces are in line with the Catena-X Style Guidelines
      - user interface style review has been executed  (review owner: SYSTEM ARCHITECT5 UX)
      - review feedback (if existing) got incorporated
      - all findings are assessed
      - all findings (high/very high) are fixed or cleaned up (evidence by re-review)
      - approval of the application CX Style conformity is available (given by the review owner)

  _Best Practice:_
    - Obtain approval from Style Guideline Owner, prior to Gate review
    - use issue, include app URL & TestUser, assign to SYSTEM ARCHITECT5, expect review loop
    - [Style Components](https://portal.dev.demo.catena-x.net/_storybook/?path=/story)
    - [LINK](https://confluence.catena-x.net/x/DVIAAQ) to Style Guideline (Source only accessible for Catena-X Consortia members in current transition phase)
    - LINK to FrontEnd validations (will be added asap)
    - Please note, you can use the [official public available CX shared component library](https://www.npmjs.com/package/cx-portal-shared-components?activeTab=readme) (react supported) to easily develop applications that are in-line with the CX style guidelines

**Comment by OSchlienz** [comment]
@vialkoje please perform the required expert review. Thanks!
[Administrator`s Guide (User assistance)](https://github.com/eclipse-tractusx/managed-identity-wallet#administrator-documentation)
[End-User Manual (User assistance)](https://github.com/eclipse-tractusx/managed-identity-wallet#end-users)
[Interfaces Documentation](https://managed-identity-wallets-new.int.demo.catena-x.net/ui/swagger-ui/index.html#/)

**Comment by OSchlienz** [comment]
**UX consistency** 
N/A: as the Managed Identity Wallet does not have an own UI. 
@jjeroch could you please confirm/approve? Thanks!

**Comment by carslen** [comment]
Potential blocker for QG4 Source Code approval: eclipse-tractusx/managed-identity-wallet#167 Requires clarification.

**Comment by borisrizov-zf** [comment]
@carslen I've left a comment in #167 about the update.

**Comment by borisrizov-zf** [comment]
@carslen I've done some dependency checks and security scans with the `cx-ssi-lib-0.0.18-SNAPSHOT` inside the `MIW`. As far as I can tell, all the necessary components are now in place, I'm working on the tasks, so we can do an official release of `cx-ssi-lib-0.0.18` and a subsequent develop release of the `MIW` which uses it.

TLDR; All the parts are in place, just have to merge some PRs now to finalise.

**Comment by OSchlienz** [comment]
@vialkoje documents for expert review, no changes in respect to R23.09
[Architecture dicuments](https://github.com/borisrizov-zf/ssi-docu/blob/main/docs/architecture/cx-3-2/2.%20Managed%20Identity%20Wallet/arc42/main.md)
[Administrator`s Guide (User assistance)](https://github.com/eclipse-tractusx/managed-identity-wallet#administrator-documentation)
[End-User Manual (User assistance)](https://github.com/eclipse-tractusx/managed-identity-wallet#end-users)
[Interfaces Documentation](https://managed-identity-wallets-new.int.demo.catena-x.net/ui/swagger-ui/index.html#/)

**Comment by carslen** [comment]
Approval subject to reservation:
- Product deployment documentation not sufficient but developer focused (`README.md`/`INSTALL.md`).
- TRG 5.11 Helm upgradeabilibity workflow missing
- Helm Chart README.md version mismatch

Small things, but needs to be fixed.

**Comment by carslen** [comment]
I have revoked my approval, as the Helm chart release for app version 0.3.0 is missing:

```shell
$ helm repo update tractusx-dev
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "tractusx-dev" chart repository
Update Complete. ⎈Happy Helming!⎈

$ helm search repo tractusx-dev/managed-identity-wallet --versions | head -n 4
NAME                                 	CHART VERSION	APP VERSION	DESCRIPTION
tractusx-dev/managed-identity-wallet 	1.1.0        	0.0.1      	Managed Identity Wallet is supposed to supply a...
tractusx-dev/managed-identity-wallet 	0.2.0        	0.2.0      	Managed Identity Wallet is supposed to supply a...
tractusx-dev/managed-identity-wallet 	0.1.0        	0.1.0      	Managed Identity Wallet is supposed to supply a...
```

@OSchlienz @borisrizov-zf You need to release the helm chart urgently! (BTW, which workflow does this job?).

**Comment by carslen** [comment]
Manually triggered GH workflow to update our Helm Chart repository. Helm repository is updated on a daily basis late in the evening.

```shell
helm search repo tractusx-dev/managed-identity-wallet --versions | head -n 4
NAME                                 	CHART VERSION	APP VERSION	DESCRIPTION
tractusx-dev/managed-identity-wallet 	1.1.0        	0.0.1      	Managed Identity Wallet is supposed to supply a...
tractusx-dev/managed-identity-wallet 	0.3.0        	0.3.0      	Managed Identity Wallet is supposed to supply a...
tractusx-dev/managed-identity-wallet 	0.2.0        	0.2.0      	Managed Identity Wallet is supposed to supply a...
```

Consider to release the helm chart as previously as a GH release too.

**Comment by borisrizov-zf** [comment]
I've added  #202 to track this.

**Comment by vialkoje** [comment]
documentation existing, reasonable content found. expert approval granted

**Comment by kelaja** [comment]
@carslen @OSchlienz can the TRGs be closed? Kindly confirm

**Comment by mabige** [comment]
All check-boxes are checked. Is this not already sufficient proof?
